src/main/java/training/endava/skeleton/Main.java

```diff
@@ -56,17 +60,131 @@ public static void main(String argv[]) {
 
         //Generics demo for lower-bound
 //        YellowFruits y = new YellowFruits();
+        ArrayList<Company> companies = (ArrayList<Company>) makeCompanies(Optional.empty());
+        companies = (ArrayList<Company>) populateCompanies(companies);
+
+        //Stream Exercises
+        //ex 1
+        companies.stream().forEach(System.out::println);
+
+        //ex2
+        //in person implementation (setId)
+
+        //ex3
+//        System.out.println(companies.stream().min((a,b)-> a.getFoundationYear().compareTo(b.getFoundationYear())).get().getName());
+
+        //ex4
+//        System.out.println(companies.stream().map(x -> x.getEmployeeList().size()).reduce(0,(a,b)->a+b));
+
+        //ex5
+//        String name = companies.stream()
+//                .map(x -> x.getEmployeeList()
+//                        .stream()
+//                        .map(y -> y.getName())
+//                        .reduce("", (a, b) -> a + b))
+//                .reduce("", (a, b) -> a + b);
+//        System.out.println(new Person(name, "email", "tel", 123456, companies.get(3)));
+
+        //ex6
+//        companies.stream().
+//                forEach(x -> x.getEmployeeList()
+//                        .stream()
+//                        .map(y -> y.getName())
+//                .forEach(System.out::println));
+        //ex7
+//        companies.stream()
+//                .forEach(x -> x.getEmployeeList()
+//                        .stream()
+//                        .map(y -> y.getPhoneNumber())
+//                        .forEach(System.out::println));
+//        System.out.println("distinct ----------------------------");
+//        companies.stream()
+//                .forEach(x -> x.getEmployeeList()
+//                        .stream()
+//                        .map(y -> y.getPhoneNumber())
+//                        .distinct()
+//                        .forEach(System.out::println));
```

Comment:
The task requested to display the persons, so to process distinct on a stream of persons, not only on the phone numbers.